### PR TITLE
Remove legacy form and other database tables/columns

### PR DESCRIPTION
**Pull request by luontola**
#1101

### CODE REVIEW

File: src/clj/rems/db/applications.clj [hunk 1]
```diff
@@ -91,10 +91,7 @@
 
 (defn create-application! [user-id catalogue-item-ids]
   (let [start (time/now)
-        app-id (:id (db/create-application! {:user user-id
-                                             ;; TODO: remove catalogue_item_application.wfid
-                                             :wfid (get-workflow-id-for-catalogue-items catalogue-item-ids)
-                                             :start start}))]
+        app-id (:id (db/create-application! {}))]
```
Comment (okahilak) [inline src/clj/rems/db/applications.clj]: minor: (db/create-application!) should work.

File: test/clj/rems/application/test_events_cache.clj [hunk 2]
```diff
@@ -5,36 +5,19 @@
...
-                                               :owneruserid ""
-                                               :modifieruserid ""
-                                               :title ""
-                                               :fnlround 1
-                                               :workflow "{}"}))
-        app-id (:id (db/create-application! {:user ""
-                                             :wfid workflow-id
-                                             :start (time/now)}))]
-    app-id))
+  (:id (db/create-application! {})))
```
Comment (okahilak) [inline test/clj/rems/application/test_events_cache.clj]: minor: ditto

File: resources/migrations/20190724113840-remove-application-form-fk.up.sql [hunk 3]
```diff
@@ -0,0 +1,5 @@
+-- begin generating form IDs using form_template_id_seq instead of application_form_id_seq
+SELECT setval('form_template_id_seq', (SELECT nextval('application_form_id_seq')), false);
+--;;
+ALTER TABLE catalogue_item
+    DROP CONSTRAINT catalogue_item_ibfk_3;
```
Comment (okahilak) [inline resources/migrations/20190724113840-remove-application-form-fk.up.sql]: Hmm, should an analogous new constraint be added that links formId to id in form_template?

Comment (luontola) [reply resources/migrations/20190724113840-remove-application-form-fk.up.sql]: That was added already earlier: https://github.com/CSCfi/rems/blob/master/resources/migrations/20190715105532-form-template-fk.up.sql

Comment (okahilak) [reply resources/migrations/20190724113840-remove-application-form-fk.up.sql]: Ok.

**Pull request merged**